### ISSUE TITLE
Proposal: Allow custom headers to be passed with each individual request

### ISSUE THREAD

**Issue by richddr** [Netflix/batch_request_api#6]
Something like:

```json
"requests": [
    {
      "method": "POST",
      "url": "/api/v1/movies",
      "body": { },
      "headers": { }
    }
  ]
}
```

**Comment by shishirmk** [comment]
@richddr are you using the batch_sequential approach to batch requests?

**Comment by richddr** [comment]
@shishirmk Yes, but I went with another implementation altogether: https://github.com/arsduo/batch_api

**Comment by sdhull** [comment]
@richddr are you using https://github.com/arsduo/batch_api in conjunction with https://github.com/Netflix/ember-batch-request or something else?

**Comment by richddr** [comment]
@sdhull just https://github.com/arsduo/batch_api, we haven't had the need to implement front-end batching of requests yet.
